samples/CMakeLists.txt

```diff
@@ -157,3 +157,5 @@ if (CONFIG_NCS_INCLUDE_RPMSG_CHILD_IMAGE)
   endif()
 
 endif()
+
+add_subdirectory(nrf9160/example_trace_medium)
```

Comment:
This looks a bit out of place.

Reply:
Yeah, I am not convinced we should have a sample showing how to add a custom trace medium.
I did not imagine the customer would use it like that, rather, they would drop a file in the `lib/nrf_modem_lib/trace` folder, add a Kconfig for it (to add a `config` to the `choice` of medium), and add one line to the CMake file there.
Then it would be available everywhere, not just in a sample but any application they want to build with NCS.

The documentation and the structure of the two mediums we provide should be good enough to guide the user in case they want to add a new trace medium.

Reply:
I agree `add_subdirectory(nrf9160/example_trace_medium)` looks out of place, but is it needed? In my opinion, the example trace medium should only be available if you build the sample with the custom trace medium.

I disagree that a customer should modify/add any files in the nrf repo for their projects. When doing an out of tree project, you'd depend on the sdk-nrf project and have all application logic in your own repo. Modifying dependencies will create a mess to maintain. That's why I think we should have a separate sample application in the samples folder to demonstrate how to add a separate trace medium. It should just be a minimal sample to make it clear what's needed to create a custom trace medium. The macro functions make it possible to inject the trace medium from anywhere, so why not make it clear by having everything that's needed in a sample folder?